interop/ksp/src/main/kotlin/com/squareup/kotlinpoet/ksp/ksTypes.kt

```diff
@@ -68,16 +66,17 @@ public fun KSType.toTypeName(
       } else {
         decl.typeParameters.toTypeParameterResolver(typeParamResolver)
       }
-      val firstPass = if (unwrapTypeAliases) {
-        decl.type.resolve()
-          .toTypeName(extraResolver, unwrapTypeAliases)
-          .copy(nullable = isMarkedNullable)
-          .rawType()
-      } else {
-        decl.toClassNameInternal()
-      }
-      firstPass
-        .withTypeArguments(arguments.map { it.toTypeName(typeParamResolver) })
+      val mappedArgs = arguments.map { it.toTypeName(typeParamResolver) }
+
+      val abbreviatedType = decl.type.resolve()
+        .toTypeName(extraResolver)
+        .copy(nullable = isMarkedNullable)
+        .rawType()
+        .withTypeArguments(mappedArgs)
+
+      decl.toClassNameInternal()
+        .withTypeArguments(mappedArgs)
+        .copy(tags = mapOf(TypeAliasTag::class to TypeAliasTag(abbreviatedType)))
```

Comment:
So we'll unwrap a typealias by default now and attach a `TypeAliasTag` to indicate that the type was unwrapped? If the use case needs typealises to not be unwrapped, would it be easy to achieve this by using the tag? Or is there a different use case here?

Reply:
Got it!